lib/xandra/connection/utils.ex

```diff
@@ -3,9 +3,11 @@ defmodule Xandra.Connection.Utils do
 
   alias Xandra.{Connection.Error, Frame, Protocol}
 
-  @spec recv_frame(:gen_tcp.socket) ::
+  @spec recv_frame(:gen_tcp.socket, nil | module) ::
         {:ok, Frame.t} | {:error, :closed | :inet.posix}
-  def recv_frame(socket) do
+  def recv_frame(socket, compressor \\ nil)
+      when is_nil(compressor)
```

Comment:
I think we can drop `when is_nil(compressor)` as it is effectively included in `is_atom(compressor)`. It rather raises question why we do so than helps.